cumulusci/tasks/util.py

```diff
@@ -68,15 +68,33 @@ def _run_task(self):
         )
 
 
-class Sleep(BaseTask):
+class Sleep(ResumableTask):
     name = "Sleep"
     task_options = {
         "seconds": {"description": "The number of seconds to sleep", "required": True}
     }
 
-    def _run_task(self):
-        self.logger.info("Sleeping for {} seconds".format(self.options["seconds"]))
-        time.sleep(float(self.options["seconds"]))
+    seconds_remaining: int = -1
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.options["seconds"] = int(self.options["seconds"])
+        self.seconds_remaining = self.options["seconds"]
+
+    def _start(self):
+        self.logger.info(f"Sleeping for {self.seconds_remaining} seconds")
+
+    def _resume(self):
+        self.logger.info(f"Continuing sleeping for {self.seconds_remaining} seconds")
+
+    def _run_step(self):
+        time.sleep(1)
+        self.seconds_remaining -= 1
+
+    def _is_finished(self):
+        return self.seconds_remaining <= 0
```

Comment:
This is useful for demonstrating the API, but I would expect an actual resumable sleep task to compare to the original start time rather than tracking how many individual seconds it has slept for. If it's supposed to sleep for 10 seconds, and actually sleeps for 2 before getting interrupted, and then gets resumed 8 seconds later, it should recognize that it has already achieved the goal of waiting 10 seconds.

Reply:
Yeah as long as you trust distributed system clocks. My way is a bit more conservative in that it really counts that all of the time passed no matter what the clocks say. For the practical uses we might use sleep for (a few seconds waiting for a process to finish?), either is probably fine?

Reply:
If sleeping were important it might make sense to bake it into a deeper layer so MetaCI could just kill the process and forget about it until the sleeping is done. Reconstituting a process to check sleep progress is kind of wasteful.

Reply:
The sleep task is mostly for testing, so it's not particularly important; most tasks that actually do need a pause just call time.sleep directly.